redhat-developer/developers.redhat.com
# Developer 5261 custom module

### JIRA Issue Link
https://issues.jboss.org/browse/DEVELOPER-5261
https://issues.jboss.org/browse/DEVELOPER-5260

### Updates
- Added new "rhd_eloqua_embed" module
- Added "Eloqua Form Embed" assembly type
- "Eloqua Form Embed" assembly available on SOLP & Landing page content types

### Verification Process
1. Create or update an SOLP content type
2. See that you can add en Eloqua form to the sidebar or to the main Sections (Sidebar not available on regular landing page content type)
3. Create a new Eloqua Form Embed assembly
4. Enter JSON from Eloqua (sample included below)
5. Save assembly and node
6. View embedded Eloqua form on sidebar or in main content

```
{
      "showAdditionalFields": [],
      "hideStandardFields": [],
      "CustomQuestions": [],
      "offer_id": "70160000000Qf1SAAS",
      "language": "en",
      "FormTitle": "Roger Test form ",
      "FormIntro": "This is a test intro paragraph!",
      "NameOrder": "western",
      "FormCallToAction": "DOWNLOAD NOW",
      "ThanksTitle": "Thank you",
      "ThanksText": "Thank you paragraph",
      "ShowThanksButton": "true",
      "leadActivity": 1,
      "disableVisitorContactLookups": false
    }
```

### Screens

#### Admin Sidebar
![screen shot 2018-09-20 at 11 03 39 am](https://user-images.githubusercontent.com/41014351/45843589-3d7a0000-bcdd-11e8-863e-a7482c4b4b5b.png)

#### Admin Sections
![screen shot 2018-09-20 at 11 03 53 am](https://user-images.githubusercontent.com/41014351/45843594-41a61d80-bcdd-11e8-9b67-1cfd92ad44c1.png)

#### SOLP
![screen shot 2018-09-20 at 1 45 11 pm](https://user-images.githubusercontent.com/41014351/45843607-4539a480-bcdd-11e8-9d23-dcab360ec5ef.png)

#### Landing Page
![screen shot 2018-09-20 at 1 44 50 pm](https://user-images.githubusercontent.com/41014351/45843616-4965c200-bcdd-11e8-97d9-8e702d2537cb.png)
